c7n/resources/org.py

```diff
@@ -191,8 +191,133 @@ def parse_query(self):
+            )
+
+            ou_assembly = hierarchy_manager.filters[0]
+            ou_assembly.ou_map = {
+                ou["Id"]: ou for ou in self.manager.get_resource_manager("org-unit").resources()
+            }
+            ou_assembly.process_accounts(resources, event)
+            # also initialize root for accounts as we dont store it as a parent.
+            self.ou_root = self.client.list_roots()["Roots"][0]
+        self.target_policies = {}
+        return super().process(resources, event)
+
+    def get_targets(self, resource):
+        if not self.data.get("inherited"):
+            yield resource["Id"]
+            return
+
+        # return self, parents, root
+
+        # handle ous
+        if self.manager.type == "org-unit":
+            yield resource["Id"]
+            for p in resource["Parents"]:
+                yield p
+            return
+
+        # handle accounts
+        yield resource["Id"]
+        for p in resource[OrgUnitFilter.annotation_parent_key]:
+            yield p["Id"]
+        yield self.ou_root["Id"]
+
+    def get_item_values(self, resource):
+        rpolicies = {}
+        for tgt_id in self.get_targets(resource):
+            if tgt_id not in self.target_policies:
+                policies = self.client.list_policies_for_target(
+                    Filter=self.data["policy-type"], TargetId=tgt_id
+                ).get("Policies", ())
+                self.target_policies[tgt_id] = policies
+            for p in self.target_policies[tgt_id]:
+                rpolicies[p["Id"]] = p
+        return list(rpolicies.values())
+
+
+@OrgAccount.action_registry.register("set-policy")
+@OrgUnit.action_registry.register("set-policy")
+class SetPolicy(Action):
+    """Set a policy on an org unit or account"""
+
+    schema = type_schema(
+        "set-policy",
+        required=["name", "policy-type"],
+        **{
+            "name": {"type": "string"},
+            "description": {"type": "string"},
+            "policy-type": {"enum": OrgPolicy.policy_types},
+            "contents": {"type": "object"},
+            "tags": {"$ref": "#/definitions/string_dict"},
+        },
+    )
+    permissions = ("organizations:AttachPolicy", "organizations:CreatePolicy")
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client("organizations")
+        pid = self.ensure_scp(client)
+        for r in resources:
+            self.manager.retry(client.attach_policy, TargetId=r["Id"], PolicyId=pid)
+
+    def ensure_scp(self, client):
+        pmanager = self.manager.get_resource_manager(
+            "org-policy", {"query": [{"filter": self.data["policy-type"]}]}
+        )
+        policies = pmanager.resources()
+        found = False
+        for p in policies:
+            if p["Name"] == self.data["name"]:
+                found = p
+                break
+        if found:
+            # todo: perhaps modify/compare to match.
```

Comment:
agreed, non matched contents should be updated? alternatively maybe error out, as that policy may be managed elsewhere?

Reply:
I could see a 2nd iteration where we allow folks to opt into an update. Otherwise yeah erroring out seems reasonable - don't want to accidentally/magically overwrite an existing policy.

Reply:
its an interesting question, I think when updating we should expect an explicit custodian tag, else log warning/error on those that don't, effectively creating an implicit tag on any policies we are managing.

Reply:
added an explicit managed-by tag to policies created by cloud custodian.